Homework27: Ekaterina Zavadskaya

KateZavadskaya opened the pull request:
Add hm27

KateZavadskaya committed:
Feat: add files to train

Add file hm27.py, 27_getUser.py

Docs:hm27.py, 27_getUser.py

KateZavadskaya committed:
Feat(hm27): add api project

Add folder config with veraibles
Add folder fixture with fixture
Add folder tests with tests

Add README.md
Add requirements.py

Docs: folders and docs.py

KateZavadskaya committed:
Feat(hm27) add hm27

KateZavadskaya committed:
Feat(hm27): fixes

Delete extra files in structure
Add correct file in strucrure
Add correct imports

KateZavadskaya committed:
Merge branch 'main' into feature/homework27

KateZavadskaya committed:
Fix: E999 SyntaxError

Fix E999 SyntaxError flake8

Doc: api/tests/test_update.py

KateZavadskaya committed:
Need to be merged Merge branch 'feature/homework25' of https://github.com/KateZavadskaya/mypy2024 into feature/homework27

Stml89 commented on api/tests/test_create.py:
```diff
@@ -0,0 +1,58 @@
+    }
+    try:
+        validate(instance=response_json, schema=schema)
+        print("JSON-ответ соответствует :) схеме")
```
use `logger` instead of `print`

Stml89 commented on api/tests/test_create.py:
```diff
@@ -0,0 +1,58 @@
+    response_json = response.json()
+    print(response.status_code)
+    print(response_json)
+    assert response.status_code == 200, \
+        f"Ожидался статус-код 200, но получен {response.status_code}"
```
move the common part to a separate module, so you can reuse it here:
```

def post(...):
    response = requests.post(...)
    assert response.status_code == expected_code, ....
    return response.json()

def get(...):
    response = requests.get(...)
    ....

def delete...
    ...

....
```

KateZavadskaya replied:
Sorry, it is more that I can for now.

Stml89 reviewed (changes requested):
Fix linters

KateZavadskaya committed:
Fix: add comments fix

Fix linters, except ruff F401, F811
Add logger in try insted of print

Docs: hm27

KateZavadskaya committed:
Fix: comments

Delete folder fixture
Created file conftest.py
Delete extra import
Fix extra agrement in test_negative.py

Docs: hm27
